File: app/src/main/java/com/pr0gramm/app/ui/intro/slides/BookmarksActionItemsSlide.java
package com.pr0gramm.app.ui.intro.slides;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

import com.google.common.collect.ImmutableList;
import com.pr0gramm.app.ActivityComponent;
import com.pr0gramm.app.R;
import com.pr0gramm.app.feed.FeedFilter;
import com.pr0gramm.app.feed.FeedType;
import com.pr0gramm.app.services.BookmarkService;

import java.util.List;

import javax.inject.Inject;

/**
 */
public class BookmarksActionItemsSlide extends ActionItemsSlide {
    @Inject
    BookmarkService bookmarkService;

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        view.setBackgroundResource(R.color.blue_primary);
    }

    @Override
    protected void injectComponent(ActivityComponent activityComponent) {
        activityComponent.inject(this);
    }

    @Override
    protected String getIntroTitle() {
        return "Lesezeichen";
    }

    @Override
    protected String getIntroDescription() {
        return "Wähle aus der Liste Lesezeichen aus, die du direkt in der Navigation sehen möchtest." +
                " Du kannst auch jederzeit weitere Lesezeichen mit eigener Suche anlegen.";
    }

    @Override
    protected List<ActionItem> getIntroActionItems() {
        FeedFilter f = new FeedFilter().withFeedType(FeedType.PROMOTED);

        return ImmutableList.of(
                new BookmarkActionItem(bookmarkService, "Kein Ton", f.withTags("? -f:sound")),
                new BookmarkActionItem(bookmarkService, "Nur Bilder", f.withTags("? -webm -gif")),
                new BookmarkActionItem(bookmarkService, "Original Content", f.withTags("original content")),
                new BookmarkActionItem(bookmarkService, "0815 & Süßvieh", f.withTags("? 0815|süßvieh|(ficken halt)|(aber schicks keinem)")),
                new BookmarkActionItem(bookmarkService, "Ton nur mit Untertitel", f.withTags("? (-f:sound | (untertitel & -404))")),
                new BookmarkActionItem(bookmarkService, "Keine Videos", f.withTags("? -webm")),
                new BookmarkActionItem(bookmarkService, "Reposts in Top", f.withTags("? repost & f:top")));
    }
}

File: app/src/main/java/com/pr0gramm/app/ui/intro/slides/BookmarksActionItemsSlide.kt
package com.pr0gramm.app.ui.intro.slides

import com.pr0gramm.app.R
import com.pr0gramm.app.feed.FeedFilter
import com.pr0gramm.app.feed.FeedType
import com.pr0gramm.app.model.bookmark.Bookmark
import com.pr0gramm.app.services.BookmarkService
import com.pr0gramm.app.util.di.instance

/**
 */
class BookmarksActionItemsSlide : ActionItemsSlide("BookmarksActionItemsSlide") {
    private val bookmarkService: BookmarkService by instance()

    override val introBackgroundResource: Int = R.color.blue_primary

    override val introTitle: String = "Lesezeichen"

    override val introDescription: String = "Wähle aus der Liste Lesezeichen aus, die du direkt " +
            "in der Navigation sehen möchtest. Du kannst auch jederzeit weitere Lesezeichen " +
            "mit eigener Suche anlegen oder vorhandene durch langes Drücken löschen."

    override val introActionItems: List<ActionItem>
        get() {
            val f = FeedFilter().withFeedType(FeedType.PROMOTED)

            return listOf(
                    BookmarkActionItem(bookmarkService, "Kein Ton",
                            f.withTags("! -f:sound")),

                    BookmarkActionItem(bookmarkService, "Community Textposts",
                            f.withTags("! 'text' & 'richtiges grau'")),

                    BookmarkActionItem(bookmarkService, "Nur Bilder",
                            f.withTags("! -'video' -'gif'")),

                    BookmarkActionItem(bookmarkService, "Original Content",
                            f.withTags("! 'original content' | 'oc'")),

                    BookmarkActionItem(bookmarkService, "Text in Top",
                            f.withTags("! 'text'")),

                    BookmarkActionItem(bookmarkService, "0815 & Süßvieh",
                            f.withTags("! 0815|süßvieh|'ficken halt'|'aber schicks keinem'")),

                    BookmarkActionItem(bookmarkService, "Ton nur mit Untertitel",
                            f.withTags("! (-f:sound | (untertitel & -404))")),

                    BookmarkActionItem(bookmarkService, "Admin & Mods",
                            f.withTags("! m:admin | m:mod")),

                    BookmarkActionItem(bookmarkService, "Zufall von früher",
                            f.withTags("! d:2012 | d:2013").withFeedType(FeedType.RANDOM)),

                    BookmarkActionItem(bookmarkService, "Reposts in Top",
                            f.withTags("! 'repost' & f:top")),

                    BookmarkActionItem(bookmarkService, "Wichteln",
                            f.withTags("! 'wichteln'")),

                    BookmarkActionItem(bookmarkService, "Nur Schrott",
                            f.withTags("! s:shit").withFeedType(FeedType.NEW)))
        }
}

val Bookmark.isAppDefaultBookmark: Boolean
    get() {
        return title in listOf(
                "Kein Ton",
                "Community Textposts",
                "Nur Bilder",
                "Original Content",
                "Text in Top",
                "0815 & Süßvieh",
                "Ton nur mit Untertitel",
                "Admin & Mods",
                "Zufall von früher",
                "Reposts in Top",
                "Wichteln",
                "Nur Schrott")
    }
